### PR TITLE
Update config on new protocol network and clean-up

### PR DESCRIPTION
**Pull request by AurelienFT**
### Summary
🛠️📝🔧

This pull request updates the protocol worker module to use the latest protocol configuration, improve the code quality and performance of the handlers, and enhance the error handling and logging of the threads. It also fixes some clippy warnings, removes some obsolete comments, and adds some documentation. The main files affected are `connectivity.rs`, `block_handler.rs`, `endorsement_handler.rs`, `operation_handler.rs`, `peer_handler.rs`, and `messages.rs`.

> _We're tuning up the protocol, we're fixing all the bugs_
> _We're handling all the errors, we're suppressing all the clugs_
> _We're naming all the threads, we're using bounded channels_
> _We're heaving on the `ProtocolConfig`, one, two, three, and pull!_

### Walkthrough
*  Remove `thread_count` field from `ProtocolConfig` struct and replace it with constant `THREAD_COUNT` ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-6288abd5afc84916487dfa91aa480aad35779a7aab0100a9ac98416eb65990a7L23-R27))
* Change type of `max_operations_per_message` field from `u64` to `u32` in `ProtocolConfig` struct ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-6288abd5afc84916487dfa91aa480aad35779a7aab0100a9ac98416eb65990a7L64-R62))
* Add new fields to `ProtocolConfig` struct for various protocol parameters and limits ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-6288abd5afc84916487dfa91aa480aad35779a7aab0100a9ac98416eb65990a7R81-R132))
* Import `ENDORSEMENT_COUNT` constant and initialize new fields of `ProtocolConfig` struct in test exports module ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-054bff72c425c79c212a96848f1d0d568325e29fbb45abd276f59b9da65d6d8fR4), [link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-054bff72c425c79c212a96848f1d0d568325e29fbb45abd276f59b9da65d6d8fR49-R73))
* Change import of `crossbeam` crate from `unbounded` to `bounded` and import `tracing::warn` macro ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-c2d72e603a7e395a81a361e7ff5aa497652841761cc6dc0c1f587abf2d778b7eL2-R2), [link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-c2d72e603a7e395a81a361e7ff5aa497652841761cc6dc0c1f587abf2d778b7eR16))
* Add clippy annotations to suppress warnings about type complexity and number of arguments in `start_connectivity_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-c2d72e603a7e395a81a361e7ff5aa497652841761cc6dc0c1f587abf2d778b7eL34-R37))
* Change creation of channels from `unbounded` to `bounded` with `ProtocolConfig` fields and create thread with name in `start_connectivity_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-c2d72e603a7e395a81a361e7ff5aa497652841761cc6dc0c1f587abf2d778b7eL55-R75))
* Change initialization of caches with `ProtocolConfig` fields and add clippy annotation to suppress warning about type complexity in `BlockCache` struct ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-c2d72e603a7e395a81a361e7ff5aa497652841761cc6dc0c1f587abf2d778b7eL74-R103))
* Add check for peer info existence and change use of `peer_id` and `addr` from reference to value in `connectivity_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-c2d72e603a7e395a81a361e7ff5aa497652841761cc6dc0c1f587abf2d778b7eL179-R192), [link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-c2d72e603a7e395a81a361e7ff5aa497652841761cc6dc0c1f587abf2d778b7eL187-R200), [link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-c2d72e603a7e395a81a361e7ff5aa497652841761cc6dc0c1f587abf2d778b7eL196-R211))
* Add error handling and create thread with name in `start_listener` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-c2d72e603a7e395a81a361e7ff5aa497652841761cc6dc0c1f587abf2d778b7eL204-R220))
* Add clippy annotation to suppress warning about large enum variant in `BlockMessage` enum ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-cd393b7f1880a5026157f50754af3696f45ba2aca45123c13dc50ce425ef33e9L10-R10), [link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-b65c06e9d8ae615cdfc4d905e87d56bddc4774cf95a759bcc678872a906d1eb3R47-R48))
* Remove redundant field name in `BlockCache` struct initialization ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-cd393b7f1880a5026157f50754af3696f45ba2aca45123c13dc50ce425ef33e9L39-R39))
* Add clippy annotation to suppress warning about number of arguments in `start_block_handler_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-8c4a38023592bc37eedf7e063e33d0fa239f88ccbc8cf279dc9941d82f026f31R45))
* Remove redundant cloning of channel sender in `start_block_handler_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-8c4a38023592bc37eedf7e063e33d0fa239f88ccbc8cf279dc9941d82f026f31L69-R70))
* Change use of `peer_id` from reference to value in `propagate_blocks` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-96858fc09a34d4fa80457fbda8ede414b0844ae40385c9c0d38574e0e6983adfL109-R109))
* Add error handling and create thread with name in `start_block_propagation_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-96858fc09a34d4fa80457fbda8ede414b0844ae40385c9c0d38574e0e6983adfL173-R190))
* Change initialization of `block_message_deserializer` with `ProtocolConfig` fields in `start_block_retrieval_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-9bb654cd720a34b029e918b26fb2b2e4ed7f43ba55a3c4d8f93511c18d4470c7L105-R117))
* Change use of `from_peer_id` from reference to value in `handle_block_message` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-9bb654cd720a34b029e918b26fb2b2e4ed7f43ba55a3c4d8f93511c18d4470c7L451-R452), [link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-9bb654cd720a34b029e918b26fb2b2e4ed7f43ba55a3c4d8f93511c18d4470c7L517-R518))
* Add clippy annotation to suppress warning about number of arguments in `start_block_retrieval_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-9bb654cd720a34b029e918b26fb2b2e4ed7f43ba55a3c4d8f93511c18d4470c7R1252))
* Add error handling and create thread with name in `start_block_retrieval_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-9bb654cd720a34b029e918b26fb2b2e4ed7f43ba55a3c4d8f93511c18d4470c7L1270-R1294))
* Remove TODO comment in `messages.rs` file of endorsement handler module ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-b10363e0cd63b1b70ba7192f1bdb2ba89931b77dd57de00c80c239752db9183eL14))
* Add clippy annotation to suppress warning about number of arguments in `start_endorsement_handler_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-1d8c8d7bfd18d2ce8b4fa7207c62858625c7f86b42d1f842f52031efcbce98f0R35))
* Add error handling and create thread with name in `start_endorsement_propagation_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-8dce794d137d73851b17676ae6c1a41db8d5d26ef53ceb9db1dd67d24b3b48b1L155-R169))
* Change initialization of `endorsement_message_deserializer` with `ProtocolConfig` fields and add error handling and empty message check in `start_endorsement_retrieval_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-c8624e622a87909c2dd0639a7cd4bde89c3919dbe979bd824c955917a7553d8bL51-R55), [link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-c8624e622a87909c2dd0639a7cd4bde89c3919dbe979bd824c955917a7553d8bL64-R70))
* Add clippy annotation to suppress warning about number of arguments in `start_endorsement_retrieval_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-c8624e622a87909c2dd0639a7cd4bde89c3919dbe979bd824c955917a7553d8bR247))
* Add error handling and create thread with name in `start_endorsement_retrieval_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-c8624e622a87909c2dd0639a7cd4bde89c3919dbe979bd824c955917a7553d8bL253-R273))
* Add documentation comments to fields of `OperationMessageDeserializerArgs` struct in `messages.rs` file of operation handler module ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-dffea73be3986ded33a9fd57d3e34ee033aa3db1f28093bdb817277ce1c1ab06L96-R106))
* Add clippy annotation to suppress warning about number of arguments in `start_operation_handler_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-62d09db317a78d54f6f233b93af261066042c5c00e3cacb2e3d5801a0143f853R35))
* Remove TODO comment in `start_operation_handler_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-62d09db317a78d54f6f233b93af261066042c5c00e3cacb2e3d5801a0143f853L48))
* Add error handling and create thread with name in `start_operation_propagation_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-ebd5ebb2124045f7a99c004e7c84a3ea6f35019c30df62f39ccc1932ce4d60bdL151-R165))
* Change initialization of `operation_message_deserializer` with `ProtocolConfig` fields and add error handling and empty message check in `start_operation_retrieval_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-6bc4c5b8233d6b37c580df3c0c9f5f2939f9ba27e89427a9aaee178e8cc14417L74-R83), [link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-6bc4c5b8233d6b37c580df3c0c9f5f2939f9ba27e89427a9aaee178e8cc14417L95-R100))
* Add clippy annotation to suppress warning about number of arguments in `start_operation_retrieval_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-6bc4c5b8233d6b37c580df3c0c9f5f2939f9ba27e89427a9aaee178e8cc14417R467))
* Add error handling and create thread with name in `start_operation_retrieval_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-6bc4c5b8233d6b37c580df3c0c9f5f2939f9ba27e89427a9aaee178e8cc14417L475-R503))
* Add clippy annotation to suppress warning about large enum variant in `PeerManagementMessage` enum and add `Default` implementation for `PeerAnnounce` struct in `messages.rs` file of peer handler module ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-3ee6b5d03a2ec8bd4ea221282881616b72acc1f9ee84cfedca9acefb431b3026R15-R16), [link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-3ee6b5d03a2ec8bd4ea221282881616b72acc1f9ee84cfedca9acefb431b3026R45))
* Change creation of channels from `unbounded` to `bounded` with `ProtocolConfig` fields and change initialization of `message_deserializer` with `ProtocolConfig` fields in `start_peer_handler_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-7165d31835e99b9c8715c9c94afb5149d2e6b8c39cb3a3dd244811f71c1ae65cL67-R74), [link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-7165d31835e99b9c8715c9c94afb5149d2e6b8c39cb3a3dd244811f71c1ae65cL80-R83))
* Change use of `peer_id` from reference to value and remove TODO comment in `handle_peer_message` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-7165d31835e99b9c8715c9c94afb5149d2e6b8c39cb3a3dd244811f71c1ae65cL97-R98), [link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-7165d31835e99b9c8715c9c94afb5149d2e6b8c39cb3a3dd244811f71c1ae65cL139-R151))
* Add documentation comments to fields of `AnnouncementDeserializerArgs` struct and add `config` field to `MassaHandshake` struct in `start_peer_handler_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-7165d31835e99b9c8715c9c94afb5149d2e6b8c39cb3a3dd244811f71c1ae65cL162-R170), [link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-7165d31835e99b9c8715c9c94afb5149d2e6b8c39cb3a3dd244811f71c1ae65cL205-R218))
* Remove TODO comment in `start_peer_tester_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-7165d31835e99b9c8715c9c94afb5149d2e6b8c39cb3a3dd244811f71c1ae65cL264-L266))
* Remove empty lines in `start_peer_tester_thread` function ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-7165d31835e99b9c8715c9c94afb5149d2e6b8c39cb3a3dd244811f71c1ae65cL323), [link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-7165d31835e99b9c8715c9c94afb5149d2e6b8c39cb3a3dd244811f71c1ae65cL338))
* Add `config` argument to `new` method of `TesterHandshake` and `Tester` structs and add clippy annotation to suppress warning about type complexity in `run` method of `Tester` struct in `tester.rs` file of peer handler module ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-9d9bbaebe84221f667ce39359894963e4c0212978110bdc7bc16fff3b7d69e09L36-R41), [link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-9d9bbaebe84221f667ce39359894963e4c0212978110bdc7bc16fff3b7d69e09R151), [link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-9d9bbaebe84221f667ce39359894963e4c0212978110bdc7bc16fff3b7d69e09L163-R171))
* Change calculation of elapsed seconds since last announcement and add error handling and create thread with name in `tester.rs` file of peer handler module ([link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-9d9bbaebe84221f667ce39359894963e4c0212978110bdc7bc16fff3b7d69e09L175-R190), [link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-9d9bbaebe84221f667ce39359894963e4c0212978110bdc7bc16fff3b7d69e09L232-R240), [link](https://github.com/massalabs/massa/pull/3875/files?diff=unified&w=0#diff-9d9bbaebe84221f667ce39359894963e4c0212978110bdc7bc16fff3b7d69e09L249-R257))

### CODE REVIEW

File: massa-protocol-worker-2/src/handlers/block_handler/messages.rs [hunk 1]
```diff
@@ -44,6 +44,8 @@ pub enum BlockInfoReply {
 }
 
 #[derive(Debug)]
+//TODO: Fix this clippy warning
+#[allow(clippy::large_enum_variant)]
```
Comment (modship) [inline massa-protocol-worker-2/src/handlers/block_handler/messages.rs:48]: should box the largest enum ?

File: massa-protocol-worker-2/src/tests/mod.rs [hunk 2]
```diff
@@ -62,7 +62,7 @@ fn basic() {
     config1.max_out_connections = 1;
     config2.initial_peers = initial_peers_file_2.path().to_path_buf();
     config2.max_in_connections = 5;
-    config2.max_out_connections = 0;
+    config2.max_out_connections = 1;
```
Comment (modship) [inline massa-protocol-worker-2/src/tests/mod.rs:65]: we keep 1 in max_out for the config2 ?

File: massa-protocol-worker-2/src/handlers/block_handler/messages.rs (hunk 1 above)
Comment (AurelienFT) [reply massa-protocol-worker-2/src/handlers/block_handler/messages.rs:48]: It's annoying to use a variant that is a Box in matches as you probably saw with the BlockMessage in tests. So for now I prefer ignore and if we find a good way in the futur remove the allow that's why I placed a todo

File: massa-protocol-worker-2/src/tests/mod.rs (hunk 2 above)
Comment (AurelienFT) [reply massa-protocol-worker-2/src/tests/mod.rs]: Removed. mb

**Review by modship — APPROVED** [review]

**Pull request merged**